eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/DisallowedWordsCheck.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DisallowedWordsCheck extends AbstractCheck {
+    private final String[] DEFAULT_VALUES = new String[] { "URL", "HTTP" };
+    private Set<String> disallowedWords = new HashSet<>(Arrays.asList(DEFAULT_VALUES));
+    private String errorMessage = "%s, All Public API Classes, Fields and Methods should follow " +
+        "Camelcase standards for the following words: %s.";
+    private boolean applyToPublic = true;
+
+    /**
+     * Adds words that Classes, Methods and Variables that should follow Camelcasing standards
+     * @param disallowedWords words that should follow normal Camelcasing standards
+     */
+    public final void setDisallowedWords(String... disallowedWords) {
+        if (this.disallowedWords != null) {
+            Collections.addAll(this.disallowedWords, disallowedWords);
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.CLASS_DEF,
+        TokenTypes.VARIABLE_DEF,
+        TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.METHOD_DEF:
+            case TokenTypes.VARIABLE_DEF:
+                String tokenName = token.findFirstToken(TokenTypes.IDENT).getText();
+                if (shouldCheckInScope(token)) {
+                    String result = getDisallowedWords(tokenName);
+                    if (result != null) {
+                        log(token, String.format(errorMessage, tokenName, this.disallowedWords.stream().collect(Collectors.joining(", ", "", ""))));
+                    }
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Should we check member with given modifiers.
+     *
+     * @param token
+     *                modifiers of member to check.
+     * @return true if we should check such member.
+     */
+    private boolean shouldCheckInScope(DetailAST token) {
```

Comment:
This might be better named 'isPublicAPI'. The method needs to check not just for PUBLIC modifier, but also for PROTECTED modifier, as either of them is considered public API that we need to guard against.